### PR TITLE
[quant][graphmode][refactor] propagateQuantizationOps

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39645 [quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing
* #39644 [quant][graphmode] Support quantization for `aten::append`
* #39615 [quant][graphmode] Fix a corner case in handling `if` in insert_observers
* #39551 [quant][graphmode] Support propagate dequantize for nodes with multiple outputs
* **#39550 [quant][graphmode][refactor] propagateQuantizationOps**

Summary:
This is to prepare for next PR that fixes propagate dequantize for ops with multiple outputs

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21942063](https://our.internmc.facebook.com/intern/diff/D21942063)

### CODE REVIEW

File: torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp [hunk 1]
```diff
@@ -826,9 +826,7 @@ void propagateQParams(
   insertDeQuantForAllUse(graph, quantized_output, quantized_output);
 }
 
-void propagateDequantize(Value* output, const std::vector<Value*> inputs) {
-  Node* n = output->node();
-  Graph* graph = n->owningGraph();
+void removeDequantizeFromInputs(const std::vector<Value*>& inputs) {
```
Comment (raghuramank100) [inline torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp:829]: Can you add a test showing the functionality here before you land?

Comment (jerryzh168) [reply torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp:829]: this is a refactor, not a new functionality, it's already tested in test_quantize_script.py

File: torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp [hunk 2]
```diff
@@ -860,26 +877,23 @@ void propagateQuantizationOps(Block* block) {
...
-      if (inputs.size() > 0) {
-        // note that we don't need to recursively check for prim::If
-        // here because if all inputs of a prim::If is dequantized
-        // the dequantize will be factored out before we get to this
-        // point
-        bool is_dequantized = true;
-        for (auto* input : inputs) {
-          is_dequantized &= input->node()->kind() == Symbol::aten("dequantize");
+    if (isSingleInputGeneralValueAtenFunction(n)) {
+      for (auto* output : n->outputs()) {
```
Comment (supriyar) [inline torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp:881]: Do we need this for loop inside the `if`, `else` statements? Can it be moved outside?

Comment (jerryzh168) [reply torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp:881]: yeah, this refactor is preparing for next PR, where we do things differently for the else branch

**Review by supriyar — APPROVED** [review]
LG

**Pull request closed without merging**